Add pennylane drawer style [unitaryhack]

MoritzWillmann opened the pull request:
**Context:**
New Style, inspired by the PennyLane design with pastel colors and "sketchy" line style.
**Description of the Change:**
Addition of two new drawing styles. Additionally some potential bug fixes in setting of rcParams of other styles.
**Benefits:**
More styles
**Possible Drawbacks:**

**Related GitHub Issues:**
#2555

albi3ro reviewed (changes requested):
Thanks for the great PR @MoritzWillmann !

All the code looks good to go, but I have some requests on naming and styling.

I think we should switch the names to `"sketch"` and `"sketch_dark"`. While we tend to use this kind of hand-drawn style in graphics and demos, "pennylane" doesn't mean anything to those who haven't been reading our demos and blogs.  I also don't want to imply this is some sort of official standardized style for PennyLane. The style is more of a fun, playful option.

Mostly the styling looks good, and I love `"patch.sketch"` effect, but I'm not a huge fan of the `patheffects.withstroke` effect.  I think the border in a different colour makes the text hard to read, and it makes the lines look less hand-drawn.  I would recommend removing the `"patch.effects"` modification.

Once that effect is removed, the `"text.color"` for the dark style would need to be inverted.  The text needs to be the opposite brightness from what's beneath it for higher contrast.  Dark things on top of bright things, and bright things on top of dark things.

If we want to go full hand-drawn style, we can also set the font, like the [xkcd style does](https://github.com/matplotlib/matplotlib/blob/v3.5.2/lib/matplotlib/pyplot.py#L658).
```
            'font.family': ['xkcd', 'xkcd Script', 'Humor Sans', 'Comic Neue',
                            'Comic Sans MS'],
```

MoritzWillmann commented:
Hey @albi3ro, thanks for the feedback!

> I think we should switch the names to `"sketch"` and `"sketch_dark"`. While we tend to use this kind of hand-drawn style in graphics and demos, "pennylane" doesn't mean anything to those who haven't been reading our demos and blogs. I also don't want to imply this is some sort of official standardized style for PennyLane. The style is more of a fun, playful option.

Agreed, done!

> Mostly the styling looks good, and I love `"patch.sketch"` effect, but I'm not a huge fan of the `patheffects.withstroke` effect. I think the border in a different colour makes the text hard to read, and it makes the lines look less hand-drawn. I would recommend removing the `"patch.effects"` modification.
> 
> Once that effect is removed, the `"text.color"` for the dark style would need to be inverted. The text needs to be the opposite brightness from what's beneath it for higher contrast. Dark things on top of bright things, and bright things on top of dark things.

I removed the `patheffects.withstroke` and changed the gate colors to darker ones to keep the white text. I think a darker text will yield issues with the title of the plot.

> If we want to go full hand-drawn style, we can also set the font, like the [xkcd style does](https://github.com/matplotlib/matplotlib/blob/v3.5.2/lib/matplotlib/pyplot.py#L658).
> 
> ```
>             'font.family': ['xkcd', 'xkcd Script', 'Humor Sans', 'Comic Neue',
>                             'Comic Sans MS'],
> ```

Honestly, I'm not a huge fan of the font, but If you prefer it I can use it.

Cheers Moritz
